docs/sections/develop_agent.rst

```diff
@@ -0,0 +1,58 @@
+
+To evaluate changes in parameters on agent performance you can run your agent against several agents with baseline parameters in the sandbox (see the quickstart on `main readme`_ or the guide in `sandbox readme`_ for details how to start the sandbox and your agent).
+
+.. _main readme: https://github.com/fetchai/agents-tac/blob/master/README.md
+
+.. _sandbox readme: https://github.com/fetchai/agents-tac/blob/master/sandbox/README.md
+
+
+Advanced: Changing the Agent's Strategy
+---------------------------------------
+
+
+An intermediate approach to developing your own agent is to adjust the strategy of your agent whilst still relying on our :class:`~tac.agents.v1.examples.baseline.BaselineAgent` implementation. This way you can focus on the decision making component of the agent relevant for the TAC.
+
+The strategy interface is defined in :class:`~tac.agents.v1.base.strategy.Strategy`. It defines the following methods:
+
+- :meth:`~tac.agents.v1.base.strategy.Strategy.supplied_good_quantities` to specify the list of quantities which are supplied by the agent.
+- :meth:`~tac.agents.v1.base.strategy.Strategy.supplied_good_pbks` to specify the set of good public keys which are supplied by the agent.
+- :meth:`~tac.agents.v1.base.strategy.Strategy.demanded_good_quantities` to specify the list of quantities which are demanded by the agent.
+- :meth:`~tac.agents.v1.base.strategy.Strategy.demanded_good_pbks` to specify the set of good public keys which are demanded by the agent.
+- :meth:`~tac.agents.v1.base.strategy.Strategy.get_proposals` to specify the proposals from the agent in the role of seller/buyer.
+
+The :meth:`~tac.agents.v1.base.strategy.Strategy.supplied_good_quantities` and :meth:`~tac.agents.v1.base.strategy.Strategy.demanded_good_quantities` methods are used to :meth:`~tac.agents.v1.base.game_instance.GameInstance.get_service_description`. The service descriptions thus generated are used for registration on the OEF. Changing these methods therefore directly affects what an agent registers and what services/goods of the agent other agents can therefore find.
+
+The :meth:`~tac.agents.v1.base.strategy.Strategy.supplied_good_pbks` and :meth:`~tac.agents.v1.base.strategy.Strategy.demanded_good_pbks` methods are used to :meth:`~tac.agents.v1.base.game_instance.GameInstance.build_services_query`. The service queries thus generated are used to search for services/goods on the OEF. Changing these methods therefore directly affects what an agent searches on the OEF.
+
+The :meth:`~tac.agents.v1.base.strategy.Strategy.get_proposals` method is used to generate proposals. Changing this method directly affects what an agent proposes. Of particular relevance here is the price at which an agent proposes to sell\buy the goods referenced in the proposal.
```

Comment:
Very good! Thank you!
I approve the PR. If you can allocate some time, it would be great to have some diagrams for every of this point (very simple ones):

- `supplied_good_quantities` & `demanded_good_quantities`: an agent that send a `register_service` to the OEF Node (both as a seller and as buyer). Before the arrow of `register_service`, there are the calls to the methods `get_service_description` and then `supplied_*`/`demanded_*` 
- `supplied_good_pbks` & `demanded_good_pbks`: an agent that sends a query to the OEF Node. Analogously to the previous point, add the calls to the relevant methods.
- `get_proposals`: just show that this method is called just after a CFP in order to generate the Propose.

Reply:
Sounds good. We can do another PR on this.